api/repositories.py

```diff
@@ -127,6 +139,17 @@ def add_house(self, user_id, name, location):
         house = self.collection.insert_one({'user_id': user_id, 'name': name, 'location': location})
         return house.inserted_id
 
+    def update_house(self, house_id, name, location):
+        house = self.get_house_by_id(house_id)
+        if house is None:
+            return False
+        if name is None:
```

Comment:
if name is None, but location is not None, you still need to update the location.
Moreover, the update shouldn't *fail* if one of the fields does not need to be updated.